cmd/serverless/main.go

```diff
@@ -276,16 +276,21 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
 
 	// start appsec
 	var (
-		appsecSubProcessor   *httpsec.InvocationSubProcessor
+		appsecSubProcessor   invocationlifecycle.InvocationSubProcessor
 		appsecProxyProcessor *httpsec.ProxyLifecycleProcessor
 	)
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		appsecSubProcessor, appsecProxyProcessor, err = appsec.New()
+		subProcessor, proxySubProcessor, err := appsec.New()
 		if err != nil {
 			log.Error("appsec: could not start: ", err)
 		}
+		if subProcessor != nil {
+			appsecSubProcessor = subProcessor
+		} else if proxySubProcessor != nil {
+			appsecProxyProcessor = proxySubProcessor
+		}
```

Comment:
When I revert this change and run the tests, I'm getting no error. This implies to me that there's still a bug in our implementation somewhere. Just because this use of the sub processor forbids nil interfaces, does not mean that someone else will not come along and use it as such.

In addition to this change, I'd lke to see something like a nil check in `InvocationSubProcessor.OnInvokeStart` and `InvocationSubProcessor.OnInvokeEnd`.